plugins/jobs/src/js/data/JobModel.ts

```diff
@@ -524,17 +527,98 @@ const fieldResolvers = {
   }
 };
 
+// Sort and filtering
+function sortJobs(
+  jobs: Job[],
+  sortBy: SortOption = "ID",
+  sortDirection: SortDirection = "ASC"
+): Job[] {
+  jobs.sort((a, b) => {
+    let result;
+
+    switch (sortBy) {
+      case "ID":
+        result = sortJobById(a, b);
+        break;
+      case "STATUS":
+        result = sortJobByStatus(a, b);
+        break;
+      case "LAST_RUN":
+        result = sortJobByLastRun(a, b);
+        break;
+      default:
+        result = 0;
+    }
+
+    const direction = sortDirection === "ASC" ? 1 : -1;
+    return result * direction;
+  });
+
+  return jobs;
+}
+
+function sortJobById(a: Job, b: Job): number {
+  return a.id.localeCompare(b.id);
+}
+
+function sortJobByStatus(a: Job, b: Job): number {
+  return (
+    JobStates[a.scheduleStatus].sortOrder -
+    JobStates[b.scheduleStatus].sortOrder
+  );
+}
+
+function sortJobByLastRun(a: Job, b: Job): number {
+  return (
+    JobStatus[a.lastRunStatus.status].sortOrder -
+    JobStatus[b.lastRunStatus.status].sortOrder
+  );
+}
+
+function filterJobs(
+  jobs: MetronomeClient.JobResponse[],
+  filter: string | null
+): MetronomeClient.JobResponse[] {
+  if (filter === null) {
+    return jobs;
+  }
+
+  return jobs.filter(({ id }) => id.indexOf(filter) !== -1);
+}
+
 export const resolvers = ({
+  fetchJobs,
   fetchJobDetail,
   pollingInterval
 }: ResolverArgs): IResolvers => ({
   Query: {
     jobs(
-      _obj = {},
-      _args: GeneralArgs = {},
+      _parent = {},
+      args: GeneralArgs = {},
       _context = {}
-    ): Observable<JobConnection | null> {
-      return Observable.of(null);
+    ): Observable<Job[] | null> {
+      const {
+        sortBy = "ID",
+        sortDirection = "ASC",
+        filter = null
+      } = args as JobsQueryArgs;
+
+      const pollingInterval$ = Observable.timer(0, pollingInterval);
+      const responses$ = pollingInterval$.switchMap(fetchJobs);
```

Comment:
Here we used `exhaustMap` in our first implementation 🤔

Reply:
We used, and then we had that discussion in the Curiosity room with Orlando. switchMap is also not the best solution here. We have to do the pooling ticket Daniel added to the sprint, and that you requested on the http-service.

Reply:
Hm, the discussion was about another delay after an already delayed response finished, not about canceling (or not) an already running request 🤔

Reply:
Yes. But both exhaust and switch have their advantages. Switch will keep dropping and never complete, but exhaust also would have older data for a while until it gets stuck.

We need the polling feature on next sprint.

Reply:
Agreed with @Fabs this is not the right place to solve it :)

Reply:
My comment wasnt meant to block this PR… 🤷‍♂️

Reply:
Are you ok to the switchMap then?

Reply:
As discussed offline with orlando: 
```
Rx.Observable
  .concat(Rx.Observable.from([1]))
  .delay(1000)
  .repeat(-1)
```